File: cslavb/Csla/DataPortal/Server/DataPortalResult.vb
Imports System.Collections.Specialized

Namespace Server

  ''' <summary>
  ''' Returns data from the server-side DataPortal to the 
  ''' client-side DataPortal. Intended for internal CSLA .NET
  ''' use only.
  ''' </summary>
  <Serializable()> _
  Public Class DataPortalResult

    Private _returnObject As Object
    Private _globalContext As HybridDictionary

    ''' <summary>
    ''' The business object being returned from
    ''' the server.
    ''' </summary>
    Public ReadOnly Property ReturnObject() As Object
      Get
        Return _returnObject
      End Get
    End Property

    ''' <summary>
    ''' The global context being returned from
    ''' the server.
    ''' </summary>
    Public ReadOnly Property GlobalContext() As HybridDictionary
      Get
        Return _globalContext
      End Get
    End Property

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    Public Sub New()

      _globalContext = ApplicationContext.GetGlobalContext

    End Sub

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    ''' <param name="returnObject">Object to return as part
    ''' of the result.</param>
    Public Sub New(ByVal returnObject As Object)

      _returnObject = returnObject
      _globalContext = ApplicationContext.GetGlobalContext

    End Sub
  End Class

End Namespace

File: cslacs/Csla/DataPortal/Server/DataPortalResult.cs
using System;
using System.Collections.Specialized;

namespace Csla.Server
{
  /// <summary>
  /// Returns data from the server-side DataPortal to the 
  /// client-side DataPortal. Intended for internal CSLA .NET
  /// use only.
  /// </summary>
  [Serializable()]
  public class DataPortalResult
  {
    private object _returnObject;
    private HybridDictionary _globalContext;

    /// <summary>
    /// The business object being returned from
    /// the server.
    /// </summary>
    public object ReturnObject
    {
      get { return _returnObject; }
    }

    /// <summary>
    /// The global context being returned from
    /// the server.
    /// </summary>
    public HybridDictionary GlobalContext
    {
      get { return _globalContext; }
    }

    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    public DataPortalResult()
    {
      _globalContext = ApplicationContext.GetGlobalContext();
    }

    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    /// <param name="returnObject">Object to return as part
    /// of the result.</param>
    public DataPortalResult(object returnObject)
    {
      _returnObject = returnObject;
      _globalContext = ApplicationContext.GetGlobalContext();
    }
  }
}
